src/mondefense.cpp

```diff
@@ -98,30 +98,32 @@ void mdefense::acidsplash( monster &m, Creature *const source,
     }
 
     size_t num_drops = rng( 4, 6 );
-    const player *const foe = dynamic_cast<player *>( source );
-    if( proj == nullptr && foe != nullptr ) {
-        if( foe->weapon.is_melee( DT_CUT ) || foe->weapon.is_melee( DT_STAB ) ) {
-            num_drops += rng( 3, 4 );
-        }
+    tripoint initial_target = m.pos();
+    if( source ) {
+        const player *const foe = dynamic_cast<player *>( source );
```

Comment:
Doing a `dynamic_cast` on a nullptr is valid. You can skip the test for `source` being non-null and just check the result of the dynamic cast instead.
```suggestion
    if( const player *const foe = dynamic_cast<player *>( source ) ) {
```